### PR TITLE
Add getPendingQueueElements

### PR DESCRIPTION
**Pull request by karlfloersch**
## Description
Add getPendingQueueElements

## Contributing Agreement

- [x] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

File: contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol [hunk 1]
```diff
@@ -248,15 +262,14 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
         override
         public
     {
-        uint40 nextQueueIndex = getNextQueueIndex();
-        uint40 queueLength = _getQueueLength();
-        _numQueuedTransactions = Math.min(_numQueuedTransactions, queueLength - nextQueueIndex);
+        _numQueuedTransactions = Math.min(_numQueuedTransactions, getNumPendingQueueElements());
```
Comment (ben-chain) [inline contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol:265]: Wouldn't this be a sign that there is a reorg in the honest sequencer case?  Could even throw if _numqueued exceeds numpending.

**Review by ben-chain — APPROVED** [review]
left a comment/question but LGTM!

File: contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol (hunk 1 above)
Comment (karlfloersch) [reply contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol:265]: Ah you’re right but this is used a little differently. So when the sequencer goes down and we need to clear the queue, the batch submitter just blindly calls this with a massive number to make sure that the queue is for sure emptied.

Comment (karlfloersch) [reply contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol:265]: And then this is also useful for normal nodes if they ever want to clear the queue without any race conditions

Comment (ben-chain) [reply contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol:265]: great, ship it then!

**Pull request merged**